pkg/model/deployment_chain.go

```diff
@@ -14,18 +14,63 @@
 
 package model
 
-import "fmt"
+func (b *ChainBlock) IsCompleted() bool {
+	switch b.Status {
+	case ChainBlockStatus_DEPLOYMENT_BLOCK_SUCCESS,
+		ChainBlockStatus_DEPLOYMENT_BLOCK_FAILURE,
+		ChainBlockStatus_DEPLOYMENT_BLOCK_CANCELLED:
+		return true
+	default:
+		return false
+	}
+}
 
-func (dc *DeploymentChain) IsSuccessfullyCompletedBlock(blockIndex uint32) (bool, error) {
-	if blockIndex >= uint32(len(dc.Blocks)) {
-		return false, fmt.Errorf("invalid block index %d given", blockIndex)
+func (b *ChainBlock) DesiredStatus() ChainBlockStatus {
```

Comment:
nit: CurrentStatus or DetermineStatus

Reply:
Lets me keep the name as is, since current status of the block needs to be update thus the name CurrentStatus is a bit not right, and the name DetermineStatus just as same as DesiredStatus from my point of view.